src/partisan_client_server_peer_service_manager.erl

```diff
@@ -401,15 +413,18 @@ establish_connections(Pending, Membership, Connections) ->
 %% keys in the dict pointing to undefined if they are disconnected or a
 %% socket pid if they are connected.
 %%
-maybe_connect({Name, _, _} = Node, Connections0) ->
-    Connections = case dict:find(Name, Connections0) of
+-spec maybe_connect(Node :: node_spec(),
+                    Connections :: partisan_peer_service_connections:t()) ->
+            partisan_peer_service_connections:t().
+maybe_connect(Node, Connections0) ->
```

Comment:
`partisan_peer_service_connections:find/2` handle both short names and node specs, the thing is it returns the first one it finds, i'm a bit unclear on what should be the result of that when multiple connections are implemented.